File: generic_py/generic_py/generic_tester_int_pub.py
from std_msgs.msg import Int64

import rclpy
from rclpy.node import Node

class GenericTesterIntPub(Node):

    def __init__(self, topic_name):
        super().__init__('generic_tester_int_pub')

        self.publisher_ = self.create_publisher(Int64, topic_name, 10)
        
        self.timer = self.create_timer(5.0, self.publish_messages)

    def publish_messages(self):
        msg = Int64()
        msg.data = 30
        
        self.get_logger().info(f"Publishing integer message type of value: '{msg.data}'")
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)

    generic_tester_pub = GenericTesterIntPub("/buffer")

    rclpy.spin(generic_tester_pub)

    generic_tester_pub.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: generic_cpp/src/generic_tester_int_pub.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int64.hpp"

using namespace std::chrono_literals;

class GenericTesterIntPub : public rclcpp::Node
{
  public:
    GenericTesterIntPub(std::string topic_name)
      : Node("generic_tester_int_pub"), topic_name_(topic_name)
    {
      int_publisher_ = this->create_publisher<std_msgs::msg::Int64>(topic_name_, 10);

      timer_ = this->create_wall_timer(5s, std::bind(&GenericTesterIntPub::publish_messages, this));
    }
  private:
    void publish_messages()
    {
      std_msgs::msg::Int64 int_msg;
      int_msg.data = 30;
      
      RCLCPP_INFO(this->get_logger(), "Publishing integer message type of value: '%ld'", int_msg.data);
      int_publisher_->publish(int_msg);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr int_publisher_;
    std::string topic_name_;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  auto node = std::make_shared<GenericTesterIntPub>("/buffer");

  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}